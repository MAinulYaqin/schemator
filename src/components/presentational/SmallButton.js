import React from "react";
import PropTypes from "prop-types";
import styled from "styled-components";
import "typeface-roboto";

const Container = styled.div`
  width: fit-content;
  padding: 4px 12px;
  background: #b33939;
  white-space: nowrap;
  cursor: pointer;
  user-select: none;

  &:hover {
    background: #8e2929;
  }
`;

const Caption = styled.span`
  font-family: Roboto;
  font-size: 12px;
  color: white;
`;

const SmallButton = ({ className, caption, onClick }) => (
  <Container className={className} onClick={onClick}>
    <Caption>{caption}</Caption>
  </Container>
);

SmallButton.propTypes = {
  className: PropTypes.string,
  caption: PropTypes.string,
  onClick: PropTypes.func
};

SmallButton.defaultProps = {
  caption: "Button"
};

export default SmallButton;
