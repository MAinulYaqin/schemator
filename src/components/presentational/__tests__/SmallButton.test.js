import React from "react";
import { shallow } from "enzyme";

import SmallButton from "../SmallButton";

describe("SmallButton", () => {
  const setup = propOverrides => {
    const props = {
      caption: "Remove",
      onClick: jest.fn(),
      ...propOverrides
    };

    const wrapper = shallow(<SmallButton {...props} />);

    return {
      wrapper,
      props
    };
  };

  it("should renders default", () => {
    const { wrapper } = setup();

    expect(wrapper).toMatchSnapshot();
  });

  it("should calls 'onClick' while clicked", () => {
    const { wrapper, props } = setup();

    wrapper.simulate("click");

    expect(props.onClick).toBeCalled();
  });
});
