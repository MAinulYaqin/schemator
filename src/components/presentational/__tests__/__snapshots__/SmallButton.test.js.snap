// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SmallButton should renders default 1`] = `
<styled.div
  onClick={[Function]}
>
  <styled.span>
    Remove
  </styled.span>
</styled.div>
`;
